### PR TITLE
405 http counter motoko example

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/motoko_examples/http_counter/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "http_counter",
+  "version": "0.1.0",
+  "description": "Reimplementation of the [Motoko Http Counter example](https://github.com/dfinity/examples/tree/master/motoko/http_counter)",
+  "main": "index.js",
+  "scripts": {
+    "test": "ts-node --transpile-only --ignore=false test/test.ts"
+  },
+  "dependencies": {
+    "azle": "^0.4.0",
```
Comment (lastmjs) [inline examples/motoko_examples/http_counter/package.json:10]: Please remove all carots `^` and merge with main and run prettier, `npx prettier --write .`

File: examples/motoko_examples/http_counter/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "http_counter",
+  "version": "0.1.0",
+  "description": "Reimplementation of the [Motoko Http Counter example](https://github.com/dfinity/examples/tree/master/motoko/http_counter)",
+  "main": "index.js",
```
Comment (lastmjs) [inline examples/motoko_examples/http_counter/package.json:5]: If you could copy the other examples' package.json files, let's remove any fields that the others do not have (like main) and set the version similar to other versions

File: examples/motoko_examples/http_counter/src/main.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+type HttpRequest = {
+    method: string;
+    url: string;
+    headers: HeaderField[];
+    body: nat8[];
+};
+
+// Look at the stable examples
+// TODO what is the stable keyword?
+let counter = 0;
```
Comment (lastmjs) [inline examples/motoko_examples/http_counter/src/main.ts:44]: Did we decide not to implement stable storage here? If we didn't, then let's do so

File: examples/motoko_examples/http_counter/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                body: encode(`Counter is ${counter}\n${req.url}\n`),
+                streaming_strategy: null,
+                upgrade: null
+            }
+        }
+        return {
+            status_code: 200,
+            headers: [ ['content-type', 'text/plain'], ['content-encoding', 'gzip'] ],
+            // body: '\1f\8b\08\00\98\02\1b\62\00\03\2b\2c\4d\2d\aa\e4\02\00\d6\80\2b\05\06\00\00\00',
+            // Currently there is an issue with azle being able to process the above string from the motoko example.
```
Comment (lastmjs) [inline examples/motoko_examples/http_counter/src/main.ts:79]: If you could prefix these comments with `TODO`, that's a convention used throughout the codebase that'll make it easy to find these. A `TODO` comment is something that should be addressed, and isn't simply there for informational purposes.

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Review by dansteren — APPROVED** [review]

**Pull request merged**